frontend/msSettingsManager.pas

```diff
@@ -480,6 +482,37 @@ procedure TSettingsManager.BuildFromPluginsItemClick(Sender: TObject);
   slSelection.Free;
 end;
 
+procedure TSettingsManager.SelectSimilarNodesItemClick(Sender: TObject);
+var
+  i, index: Integer;
+  tvNode: TTreeNode;
+  reqSmashType, currentSmashType: TSmashtype;
+  slSelection: TStringList;
+begin
+  slSelection := TStringList.Create;
+  try
+    for i := 0 to Pred(tvRecords.SelectionCount) do
+    begin
+      tvNode := tvRecords.Selections[i];
+      reqSmashType := TElementData(tvNode.Data).smashType;
+      slSelection.AddObject(tvNode.Text, TObject(reqSmashType));
+    end;
+    for i := 0 to Pred(tvRecords.Items.Count) do begin
+      currentSmashType := TElementData(tvRecords.Items[i].Data).smashType;
+      index := slSelection.IndexOf(tvRecords.Items[i].Text);
```

Comment:
If you're going to reference `tvRecords.Items[i]` twice, why not put it into your temporarily variable `tvNode`?